src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -295,14 +296,18 @@ impl ThinPool {
 
                 let usage = &status.usage;
                 if usage.used_meta > usage.total_meta - META_LOWATER {
-                    // TODO: Extend meta device
+                    match self.extend_thinpool_meta(dm, usage.total_meta, bd_mgr) {
```

Comment:
By passing `usage.total_meta` we're setting the meta extension policy to be "double the size". I think it might not be super obvious this is where that policy is being enacted, so maybe add a comment explicitly stating this, or even a line `let meta_extend_size = usage.total_meta` and then pass `meta_extend_size` could make it clear this is intentional.

Reply:
ok - sounds good